OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/CourseDiscussionResponsesAdapter.java

```diff
@@ -164,29 +168,34 @@ public void run() {
                     "cohort", groupName));
         }
 
-        bindSocialView(holder.socialLayoutViewHolder, discussionThread);
-
         bindNumberResponsesView(holder.numberResponsesViewHolder);
 
-        holder.discussionReportViewHolder.reportLayout.setOnClickListener(new View.OnClickListener() {
-            public void onClick(final View v) {
-                discussionService.setThreadFlagged(discussionThread.getIdentifier(),
-                        new FlagBody(!discussionThread.isAbuseFlagged()))
-                        .enqueue(new ErrorHandlingCallback<DiscussionThread>(
-                                context,
-                                CallTrigger.USER_ACTION,
-                                (TaskProgressCallback) null) {
-                            @Override
-                            protected void onResponse(@NonNull final DiscussionThread topicThread) {
-                                discussionThread = discussionThread.patchObject(topicThread);
-                                notifyItemChanged(0);
-                                EventBus.getDefault().post(new DiscussionThreadUpdatedEvent(discussionThread));
-                            }
-                        });
-            }
-        });
-
-        holder.discussionReportViewHolder.setReported(discussionThread.isAbuseFlagged());
+        final String loggedInUser = loginPrefs.getUsername();
+        if (loggedInUser != null && loggedInUser.equals(discussionThread.getAuthor())) {
```

Comment:
Is there any valid case where either of these values may be `null` here? I suppose the `null` check here is in response to the `@Nullable` annotation on the `getUsername()` method, which is present since it may be called before login or such. In which case I would suggest that you reverse the comparison, and thereby get rid of the unneeded `null` check. Same goes for the other sections as well.